src/commands/createFunctionApp/stacks/getStackPicks.ts

```diff
@@ -71,6 +82,16 @@ export async function getStackPicks(context: IFunctionAppWizardContext): Promise
         }
     }
 
+    if (hasEndOfLife) {
+        picks.push({
+            label: localize('endOfLife', `$(extensions-warning-message) Some stacks have an end of support deadline coming up. Learn more...`),
+            onPicked: async () => {
+                await openUrl('https://learn.microsoft.com/en-us/azure/azure-functions/functions-versions?tabs=azure-cli%2Cwindows%2Cin-process%2Cv4&pivots=programming-language-csharp://aka.ms/func-end-of-life');
+            },
+            data: { stack: stacks[0], majorVersion: stacks[0].majorVersions[0], minorVersion: stacks[0].majorVersions[0].minorVersions[0] },
```

Comment:
Hmm... seems sorta weird to include this here just to satisfy the type of picks. I think we should let the caller handle adding this pick instead of adding it here inside of `getStackPicks`. This pick isn't a stack and is just for UX purposes.

Reply:
Follow up: I think you can change the return type of `getStackPicks` to `Promise<IAzureQuickPickItem<FullFunctionAppStack | undefined>[]>` so you don't have to leave data as undefined here.